parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnWriteStoreBase.java

```diff
@@ -190,13 +190,18 @@ public void endRecord() {
 
   private void sizeCheck() {
     long minRecordToWait = Long.MAX_VALUE;
+    long maxUnwrittenRows = 0;
```

Comment:
The logic using maxUnwrittenRows is confusing to me. What does it mean for a row to be "unwritten"?

I think this should calculate the next size check row count more directly.